### PR TITLE
Update for raiding-enabled Groups

### PR DESCRIPTION
**Pull request by apv1313**
Necessary changes:
- Store info about being in PVP and recent attackers even after player death
- "PVP tag"

Proposed changes:
- Jester's (ir)regular, random, unannounced end resets (now moved to a different PR)
- PVP protection is disabled early by command rather than by hitting another player
- Endermen drop pearls normally in the end, but only on the main island

### CODE REVIEW

**Comment by apv1313** [comment]
java.time which I have been using for the end resets only exists since Java 8, which is why the Travis build (using version 7) fails. Is this a problem?

**Comment by sbezboro** [comment]
> java.time which I have been using for the end resets only exists since Java 8, which is why the Travis build (using version 7) fails. Is this a problem?

Just set travis to build oraclejdk8. Please rebase on master

**Comment by apv1313** [comment]
If incorporating the end reset changes, you'll also need to add the title "dragon-slayer" to config. (Proposed - displayName: Dragon Slayer, hidden: false, broadcast: true)

File: src/com/sbezboro/standardplugin/commands/EndresetCommand.java [hunk 1]
```diff
@@ -22,7 +23,16 @@ public boolean handle(CommandSender sender, Command command, String label, Strin
 		
 		if (plugin.getEndResetManager().isEndResetScheduled()) {
 			long endReset = plugin.getEndResetStorage().getNextReset();
-			sender.sendMessage(ChatColor.BLUE + "The next end reset is on " + ChatColor.AQUA + MiscUtil.friendlyTimestamp(endReset));
```
Comment (sbezboro) [inline src/com/sbezboro/standardplugin/commands/EndresetCommand.java:25]: can you put the changes related to end resets in another PR? I want to deploy the changes here with just the PVP changes for now. End reset stuff can be disabled after the reset until things are more stable

File: src/com/sbezboro/standardplugin/listeners/PlayerInteractListener.java [hunk 2]
```diff
@@ -40,6 +40,9 @@ public void onPlayerInteract(final PlayerInteractEvent event) {
 		} else if (event.getAction() == Action.RIGHT_CLICK_BLOCK && clickedBlock.getType() == Material.CHEST) {
 			StandardPlayer player = plugin.getStandardPlayer(event.getPlayer());
 			plugin.getHoneypotManager().checkChest(clickedBlock.getLocation(), player);
+		// Ender crystal handling
+		} else if (event.getAction() == Action.RIGHT_CLICK_BLOCK && itemStack != null && itemStack.getType() == Material.END_CRYSTAL) {
+			event.setCancelled(true);
```
Comment (sbezboro) [inline src/com/sbezboro/standardplugin/listeners/PlayerInteractListener.java]: this is for disabling dragon respawn, correct?

File: src/com/sbezboro/standardplugin/model/StandardPlayer.java [hunk 3]
```diff
@@ -297,15 +314,14 @@ public void setInPvp(StandardPlayer victim) {
 		pvpTimerTask = new PvpTimerTask(StandardPlugin.getPlugin(), this);
 		pvpTimerTask.runTaskLater(StandardPlugin.getPlugin(), PVP_TIMER_TIME);
 		
-		lastAttacker = victim;
+		lastAttackerUuid = new String(opponent.getUuidString());
```
Comment (sbezboro) [inline src/com/sbezboro/standardplugin/model/StandardPlayer.java:311]: dont need `new String()`

**Comment by sbezboro** [comment]
If the changes related to end resets are moved to a separate PR, this will be much more easier to review and deploy 👍

**Pull request merged**